### PR TITLE
Change the JSON stream API

### PR DESCRIPTION
**Pull request by samoht**
The previous API was hard to parse (#266) and wrong (#269). This commit tries to
fix that.

Now the stream is an (almost valid) JSON array which looks like:

[ {"stream":"start"}, {"version":"..."}, {"result":..},..,{"result":..},{"stream":"end"}]

almost valid as the spec doesn't allow multiple fields with the same name; but
most of existing implementation don't really care.

This needs a special release of ezjsonm (0.4.2)

### CODE REVIEW

**Comment by MagnusS** [comment]
ok - I'll update the wiki :)

**Comment by dsheets** [comment]
Where are the duplicated fields?

The reason the spec warns against duplicated fields is that no definite semantics for field lookup in the face of duplicate entries could be decided. Is it lexically first? Lexically last? Object field order doesn't matter otherwise and isn't recoverable from, e.g., `JSON.parse`.

**Comment by samoht** [comment]
well you have a new `result` field every-time you get a new element of the stream.

**Comment by dsheets** [comment]
I don't think that violates JSON if the container is still an array. Is there some other spec you're talking about?

**Comment by dbuenzli** [comment]
> well you have a new result field every-time you get a new element of the stream.

You need to get some sleep.

**Comment by samoht** [comment]
That was a good advice :-) sorry for the noise ...
